src/dpl/src/Opendp.cpp

```diff
@@ -377,6 +377,58 @@ void Opendp::groupInitPixels2()
   }
 }
 
+dbInst* Opendp::getAdjacentInstance(dbInst* inst, bool left) const
+{
+  const Rect core = grid_->getCore();
+  const Rect inst_rect = inst->getBBox()->getBox();
+  GridX left_x = grid_->gridX(DbuX{inst_rect.xMin() - 1 - core.xMin()});
+  GridX right_x = grid_->gridX(DbuX{inst_rect.xMax() + 1 - core.xMin()});
+
+  GridY y = grid_->gridSnapDownY(DbuY{inst_rect.yMin() - core.yMin()});
+
+  Pixel* pixel
+      = left ? grid_->gridPixel(left_x, y) : grid_->gridPixel(right_x, y);
+
+  dbInst* adjacent_inst = nullptr;
+
+  if (pixel != nullptr) {
+    // do not return macros, endcaps and tapcells
+    if (pixel->cell && pixel->cell->db_inst_->isCore()) {
+      adjacent_inst = pixel->cell->db_inst_;
+    }
+  }
+
+  return adjacent_inst;
+}
+
+std::vector<dbInst*> Opendp::getAdjacentInstancesCluster(dbInst* inst) const
+{
+  const bool left = true;
+  const bool right = false;
+  std::vector<dbInst*> adj_inst_cluster;
+  adj_inst_cluster.push_back(inst);
+
+  dbInst* left_inst = getAdjacentInstance(inst, left);
+  while (left_inst != nullptr) {
+    adj_inst_cluster.push_back(left_inst);
+    // the right instance can be ignored, since it was added in the line above
+    left_inst = getAdjacentInstance(left_inst, left);
+  }
+
+  dbInst* right_inst = getAdjacentInstance(inst, right);
+  while (right_inst != nullptr) {
+    adj_inst_cluster.push_back(right_inst);
+    // the left instance can be ignored, since it was added in the line above
+    right_inst = getAdjacentInstance(right_inst, right);
+  }
+
+  auto cmp = [](const dbInst* inst1, const dbInst* inst2) {
+    return inst1->getLocation().getX() < inst2->getLocation().getX();
+  };
+  std::sort(adj_inst_cluster.begin(), adj_inst_cluster.end(), cmp);
```

Comment:
No need to sort - push the left elements, reverse them, push the middle, then push the right elements.